oak_sign/src/tests.rs

```diff
@@ -0,0 +1,49 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use assert_matches::assert_matches;
+use oak_sign::{get_sha256_hex, KeyPair, SignatureBundle};
+
+const TEST_DATA: &str = "Test data";
+const TEST_SHA256: &str = "e27c8214be8b7cf5bccc7c08247e3cb0c1514a48ee1f63197fe4ef3ef51d7e6f";
+
+#[test]
+fn key_pair_test() {
+    let generate_result = KeyPair::generate();
+    assert_matches!(generate_result, Ok(_));
+    let key_pair = generate_result.unwrap();
+
+    let pkcs8_key_pair = key_pair.pkcs8_key_pair();
+    let parse_result = KeyPair::parse(&pkcs8_key_pair);
+    assert_matches!(parse_result, Ok(_));
+}
+
+#[test]
+fn signature_test() {
+    let key_pair = KeyPair::generate().expect("Couldn't create key pair");
+    let create_result = SignatureBundle::create(TEST_DATA.as_bytes(), &key_pair);
+    assert_matches!(create_result, Ok(_));
+    let signature = create_result.unwrap();
+
+    let verify_result = signature.verify();
+    assert_matches!(verify_result, Ok(_));
```

Comment:
also verify an invalid signature, or create two signature with different keys and check only the corresponding ones check correctly

Reply:
Added an incorrect signature

Reply:
Cool thanks! Though personally I still think something like this would read nicely:

```rust
let key_0 = ...
let key_1 = ...

let sign_0 = ...
let sign_1 = ...

assert_eq!(true, verify(sign_0, key_0))
assert_eq!(false, verify(sign_0, key_1))
assert_eq!(false, verify(sign_1, key_0))
assert_eq!(true, verify(sign_1, key_1))
```